nixos/doc/manual/release-notes/rl-2411.section.md

```diff
@@ -18,6 +18,8 @@
 
 ## Backward Incompatibilities {#sec-release-24.11-incompatibilities}
 
+- `gnome.nautilus` was renamed to `nautilus` and is now expected to be managed through [](#opt-programs.nautilus.enable) NixOS option. Additionally, binary Nautilus extensions will no longer be picked up from [](#opt-environment.systemPackages), they need to be installed through [](#opt-programs.nautilus.extensions). (This does not apply to `nautilus-python` scripts, which should be installed through as before.)
```

Comment:
Need to decide what to do about packages `file-roller` like or `gnome-terminal`. People might be no be aware that the Nautilus features come from those packages and be confused when they disappear.

Though those two have NixOS modules so perhaps we could set `programs.nautilus.extensions` unconditionally. It feels a bit iffy but that is the nature of one package extending another. It also would not be possible to disable just the Nautilus extension but that is not different from status quo.

Reply:
Or maybe we could adjust the patch to load extensions both from the directory specified in the environment variable and from `/run/current-system/sw/…`. That would avoid the BC break.

Reply:
I'm a bit confused, on the one hand, there is the idea of `symlinkJoin` which puts many extensions together, and on the other hand there is already `NAUTILUS_4_EXTENSION_DIR` which works already. Why do we need `symlinkJoin` to begin with?

Reply:
`NAUTILUS_4_EXTENSION_DIR` requires a single path. Previously, the `symlinkJoin` was just done by `environment.systemPackages`.

Reply:
> `NAUTILUS_4_EXTENSION_DIR` requires a single path.

Still, you can revert the `symlinkJoin` change, and use the `programs.nautilus.extensions` to simply put packages' paths in `$NAUTILUS_4_EXTENSION_DIR`.

Reply:
Each package will be in a different store path so we cannot put them all into the variable that expects a single path.

Reply:
> Each package will be in a different store path so we cannot put them all into the variable that expects a single path.

Yes but what still, in the module level, you can symlinkJoin the extensions to a single path and use that as the value for `$NAUTILUS_4_EXTENSION_DIR`.

Reply:
Well, one of the points is that we want to avoid relying on NixOS module, so that Nautilus with extensions can be used e.g. in nix-shell or h-m.